autoload/lsp.vim

```diff
@@ -568,16 +568,37 @@ function! s:on_exit(server_name, id, data, event) abort
     endif
 endfunction
 
+let s:message_types = {1: 'Error', 2: 'Warning', 3: 'Info', 4: 'Log'}
+
+function! s:handle_notification(server_name, data) abort
+    let l:response = a:data['response']
+    let l:method = l:response['method']
+    let l:params = l:response['params']
+
+    if l:method ==# 'textDocument/publishDiagnostics'
+        if g:lsp_diagnostics_enabled
+            call lsp#ui#vim#diagnostics#handle_text_document_publish_diagnostics(a:server_name, a:data)
+        endif
+    elseif l:method ==# 'language/status'
```

Comment:
I don't see this as part of the LSP spec. Can we remove it?
 One can easily register custom notifications by using `call lsp#register_notification('java', {notification->echo "notification"})`.

We might want to update `lsp#register_notifications` so that we can add multiple callbacks as well as a way to unregister.

If java people want to make it easy for them, I would expect them to go and create a plugin that does it automatically for them and share the repo. This allows the community to add custom commands such as maven install and build without updating vim-lsp such as `:MvnInstall`. The goal is to let vim-lsp be the core to enable 

```vim
Plug 'user/java-vim-lsp'
```

Reply:
Probably the register_notification should take method name?

```
call register_notification('language/status', {notification->execute('echo notification')})
```

Reply:
First param was name so that one can call register and unregister. Now that I think of it seems like it should had just returned a random id.

```vim
let id = lsp#register_notifications(function('s:notification'))
call lsp#unregister_notifications(id)
```

This including your proposed idea will be breaking change but I think it is ok to break here. It isn't a popular. We could use `*` for all messages.

```vim
call lsp#register_notification('*', {notification->exec('echo notification')})
```

But this brings up questions about how do I listen to all messages but only a particular server. Hence I was leaning to only have a callback and the logic needs to be in the callback.

Don't think it is valid vim code but if something like this worked it would be good.
```vim
call lsp#register_notifications({n-> if n.method == 'language/status' | exec('echo notification')  | endif })
```

Having just callback would allow anyone to have any logic but it would no longer be one line. So tradeoffs here.

Reply:
Looks nice. So the callback should return 0 or 1 to indicate `handled` or `continue`?